SYCL/InvokeSimd/Regression/nbarrier_multiple.cpp

```diff
@@ -0,0 +1,114 @@
+// TODO: enable on Windows once driver is ready
+// NOTE: named barrier supported only since GEN12
+// REQUIRES: gpu-intel-gen12 && linux
+// UNSUPPORTED: cuda || hip
+//
+// TODO: enable when Jira issue resolved
+// XFAIL: gpu-intel-gen12
+//
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr -std=c++20 -DXETPP_NEW_XMAIN -DUSE_OPENBLAS %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+//
+// VISALTO enable run
+// RUN: env IGC_VISALTO=63 IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+/*
+ * Test check support for named barriers in invoke_simd.
+ * This case contains two such barriers.
+ */
+
+#define XETPP_CODE_BASE __ESIMD__
```

Comment:
do we need these two defines?

Reply:
They are artifacts left when porting the tests. Removed.